WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListViewModel.kt

```diff
@@ -15,25 +22,86 @@ class PublicizeListViewModel @Inject constructor(
     private val eventBusWrapper: EventBusWrapper,
     @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
 ) : ScopedViewModel(bgDispatcher) {
+    private val _uiState = MutableLiveData<UIState>()
+    val uiState: LiveData<UIState> = _uiState
+
+    private val _actionEvents = SingleLiveEvent<ActionEvent>()
+    val actionEvents: LiveData<ActionEvent> = _actionEvents
+
+    private var shouldUpdateServices = true
+    private var twitterPublicizeService: PublicizeService? = null
+
     fun onSiteAvailable(siteModel: SiteModel) {
+        if (!shouldUpdateServices) {
+            return
+        }
+        shouldUpdateServices = false
         val siteId = siteModel.siteId
         updateServices(siteId)
-        updateConnections(siteId)
+    }
+
+    fun onTwitterDeprecationNoticeItemClick() {
+        twitterPublicizeService?.let {
+            _actionEvents.value = ActionEvent.OpenServiceDetails(it)
+        }
     }
 
     private fun updateServices(siteId: Long) {
         publicizeUpdateServicesV2.updateServices(
             siteId = siteId,
-            success = { eventBusWrapper.post(PublicizeEvents.ConnectionsChanged()) },
-            failure = { AppLog.e(AppLog.T.SHARING, "Error updating publicize services", it) }
+            success = { services ->
+                services.forEach {
+                    if (it.id == PublicizeService.TWITTER_SERVICE_ID) {
+                        twitterPublicizeService = it
+                    }
+                }
+                updateConnections(siteId)
+            },
+            failure = {
+                AppLog.e(AppLog.T.SHARING, "Error updating publicize services", it)
+                updateConnections(siteId)
+            }
         )
     }
 
     private fun updateConnections(siteId: Long) {
         publicizeUpdateServicesV2.updateConnections(
             siteId = siteId,
-            success = { eventBusWrapper.post(PublicizeEvents.ConnectionsChanged()) },
+            success = { connections ->
+                val isTwitterDeprecated = twitterPublicizeService?.status == PublicizeService.Status.UNSUPPORTED
+                val twitterConnection = connections.find {
+                    it.service == PublicizeService.TWITTER_SERVICE_ID
+                            && it.status == PublicizeConnection.ConnectStatus.OK.toString()
+                }
+                if (isTwitterDeprecated && twitterConnection != null) {
+                    _uiState.value = UIState.ShowTwitterDeprecationNotice(
+                        title = twitterConnection.label,
+                        description = R.string.sharing_twitter_deprecation_notice_description,
+                        findOutMore = R.string.sharing_twitter_deprecation_notice_find_out_more,
+                        findOutMoreUrl =
+                        "https://wordpress.com/blog/2023/04/29/why-twitter-auto-sharing-is-coming-to-an-end/",
```

Comment:
💡 Suggestion: this could be either a constant or a non-translatable string resource, though I prefer the constant in this scenario since it's being emitted from a `ViewModel` and doesn't make much sense to pass that from this context, even being non-translatable.

Reply:
Makes sense! Extracted to constant, thanks.